test/formal/opcodes.py

```diff
@@ -56,3 +56,7 @@ def SHR(x, y):
 
 def SAR(x, y):
 	return y >> x
+
+def BYTE(i, x):
+	bit = (i + 1) * 8
+	return If(UGT(bit, x.size()), BitVecVal(0, x.size()), (LShR(x, (x.size() - bit))) & 0xff)
```

Comment:
I really like these z3 rule proofs in general - but funnily it now boils down to really make sure that definitions like this here are correct :-D. But it looks good!

Reply:
Lets a proof with i=[0, 33] with expected inputs and outputs :)

Reply:
The original plan was to run the opcode definitions here against the ethereum test suite or something like that :-).